src/engine/authentication.md

```diff
@@ -0,0 +1,52 @@
+# Authentication
+
+The `engine` JSON-RPC interface, exposed by EL and consumed by CL, needs to be authenticated. The authentication scheme chosen for thus purpose is [JWT](https://jwt.io/). 
+
+The type of attacks that this authentication scheme attempts to protect against are the following: 
+
+- RPC port exposed towards the internet, allowing attackers to exchange messages with EL engine api. 
+- RPC port exposed towards the browser, allowing malicious webpages to submit messages to the EL engine api. 
+
+The authentication scheme is _not_ designed to 
+
+- Prevent attackers with capability to read ('sniff') network traffic from reading the traffic, 
+- Prevent attackers with capability to read ('sniff') network traffic from performing replay-attacks of earlier messages. 
+
+Authentication is performed as follows: 
+
+- For `HTTP` dialogue, each `jsonrpc` request is individually authenticated by supplying `JWT` token in the HTTP header. 
+- For a WebSocket dialogue, only the initial handshake is authenticated, after which the message dialogue proceeds without further use of JWT.
+  - Clarification: The websocket handshake starts with the client performing a websocket upgrade request. This is a regular http GET request, and the actual
+parameters for the WS-handshake are carried in the http headers. 
+- For `inproc`, a.k.a raw ipc communication, no authentication is required, under the assumption that a process able to access `ipc` channels for the process, which usually means local file access, is already sufficiently permissioned that further authentication requirements do not add security. 
+
+
+## JWT specifications
+
+- Client software MUST expose the authenticated Engine API at a port independent from legacy JSON-RPC API.
+  - The default port for the authenticated Engine API is `8551`. The Engine API is exposed under the `engine` namespace.
+- On the authenticated endpoint, the legacy API **MUST** also be available.
```

Comment:
Is this a reference to the JSON-RPC API? Do we want to expose an entire set of JSON-RPC on the same port? The recent idea was to restrict it to a subset of `eth_` methods that are required for CL to track deposit contract and terminal block.

Reply:
> Do we want to expose an entire set of JSON-RPC on the same port? 

I thought the CL would require most of the things in `eth` - but yes, we could restrict it. However, it would be cumbersome if the CL needs to speak to two different ports, one without auth and one with.

Reply:
Sure, but I think we should list a subset of methods required by CL in the main Engine API specification. And we may not mention it here at all as there will be only one Engine API endpoint which is an authenticated one.

Reply:
Can do this in a separate pr